tests/Unleash.Tests/Internal/ErrorEvents_Tests.cs

```diff
@@ -0,0 +1,273 @@
+﻿using FakeItEasy;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http.Headers;
+using System.Net.Http;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+using static Unleash.Tests.Specifications.TestFactory;
+using Unleash.Tests.Mock;
+using Unleash.Scheduling;
+using System.Threading;
+using Unleash.Internal;
+using Unleash.Events;
+using Unleash.Communication;
+using Unleash.Serialization;
+using FluentAssertions;
+using Unleash.Metrics;
+using System.IO;
+
+namespace Unleash.Tests.Internal
+{
+    public class ErrorEvents_Tests
+    {
+        [Test]
+        public void Fetch_Toggles_Unauthorized_Raises_ErrorEvent()
+        {
+            // Arrange
+            ErrorEvent callbackEvent = null;
+            var fakeHttpMessageHandler = new TestHttpMessageHandler()
+            {
+                Response = new HttpResponseMessage(HttpStatusCode.Unauthorized) { Content = new StringContent("Unauthorized", Encoding.UTF8) },
+            };
+            var httpClient = new HttpClient(fakeHttpMessageHandler) { BaseAddress = new Uri("http://localhost") };
+            var callbackConfig = new EventCallbackConfig()
+            {
+                ErrorEvent = evt => { callbackEvent = evt; }
+            };
+            var unleashClient = new UnleashApiClient(httpClient, new DynamicNewtonsoftJsonSerializer(), new UnleashApiClientRequestHeaders(), eventConfig: callbackConfig);
+            var cancellationTokenSource = new CancellationTokenSource();
+
+            // Act
+            var result = unleashClient.FetchToggles("123", cancellationTokenSource.Token).Result;
+
+            // Assert
+            callbackEvent.Should().NotBeNull();
```

Comment:
I think it'd be nice if these tests actually checked that the response was correct as well. This should help us catch cases where we bubble stupid/incorrect events

Reply:
Ah yes thank you, unintentional omission. Will do